src/Slackbot.Net.Extensions.FplBot/RecurringActions/GameweekMonitorRecurringAction.cs

```diff
@@ -0,0 +1,228 @@
+﻿using Fpl.Client.Abstractions;
+using Fpl.Client.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Slackbot.Net.Abstractions.Hosting;
+using Slackbot.Net.SlackClients.Http;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Slackbot.Net.Extensions.FplBot.RecurringActions
+{
+    internal class GameweekMonitorRecurringAction : GameweekRecurringActionBase
+    {
+        private readonly IFixtureClient _fixtureClient;
+        private readonly ITeamsClient _teamsClient;
+        private readonly IPlayerClient _playerClient;
+
+        private ICollection<Fixture> _currentGameweekFixtures;
+
+        public GameweekMonitorRecurringAction(
+            IOptions<FplbotOptions> options,
+            IGameweekClient gwClient,
+            ILogger<GameweekMonitorRecurringAction> logger,
+            ITokenStore tokenStore,
+            ISlackClientBuilder slackClientBuilder,
+            IFixtureClient fixtureClient,
+            ITeamsClient teamsClient,
+            IPlayerClient playerClient
+            ) : base(options, gwClient, logger, tokenStore, slackClientBuilder)
+        {
+            _fixtureClient = fixtureClient;
+            _teamsClient = teamsClient;
+            _playerClient = playerClient;
+        }
+
+        protected override async Task DoStuffWhenInitialGameweekHasJustBegun(int newGameweek)
+        {
+            await Reset(newGameweek);
+        }
+
+        protected override async Task DoStuffWhenNewGameweekHaveJustBegun(int newGameweek)
+        {
+            await Reset(newGameweek);
+        }
+
+        private async Task Reset(int newGameweek)
+        {
+            _currentGameweekFixtures = await _fixtureClient.GetFixturesByGameweek(newGameweek);
+        }
+
+        protected override async Task DoStuffWithinCurrentGameweek(int currentGameweek, bool isFinished)
+        {
+            if (isFinished)
+            {
+                return;
+            }
+
+            var newGameweekFixtures = await _fixtureClient.GetFixturesByGameweek(currentGameweek);
+
+            var players = await _playerClient.GetAllPlayers();
+            var teams = await _teamsClient.GetAllTeams();
```

Comment:
Players and teams can be fetched at gameweek start instead of each minute-cycle, so I suggest you move it to the Reset method. Also, they can be done i parallell, instead of one waiting for the other:
```csharp
1 var playersTask = _playerClient.GetAllPlayers();
2 var teamsTask = _teamsClient.GetAllTeams();
3 var players = await playersTask;
4 var teams = await teamsTask;
```
☝️  writing it this way fires the async tasks on individual threads (line 1 and 2), so that when you need the teams (line 4), the task will probably already be finished.